# _Physician Compare National_

This notebook is a continuation from of my analysis on the following data gathered via [Data.Medicare.gov](https://data.medicare.gov/Physician-Compare/Physician-Compare-National-Downloadable-File/mj5m-pzi6). It contains general information about individual eligible professionals (EPs) such as demographic information and Medicare quality program participation. This dataset is updated twice a month with the most current demographic information available at that time.

In [1]:
# import pandas
import pandas as pd
pd.options.display.max_columns = None

In [3]:
# import CSV file with data
data = pd.read_csv('physician_compare_national.csv'); 

In [4]:
data.head()

npi  ind_pac_id      ind_enrl_id     lst_nm  frst_nm mid_nm suff  \
0  1003000126  7517003643  I20130530000085  ENKESHAFI  ARDALAN    NaN  NaN   
1  1003000126  7517003643  I20130530000085  ENKESHAFI  ARDALAN    NaN  NaN   
2  1003000126  7517003643  I20150824000105  ENKESHAFI  ARDALAN    NaN  NaN   
3  1003000126  7517003643  I20150824000105  ENKESHAFI  ARDALAN    NaN  NaN   
4  1003000126  7517003643  I20150824000105  ENKESHAFI  ARDALAN    NaN  NaN   

  gndr cred med_sch  grd_yr           pri_spec sec_spec_1 sec_spec_2  \
0    M  NaN   OTHER  1994.0  INTERNAL MEDICINE        NaN        NaN   
1    M  NaN   OTHER  1994.0  INTERNAL MEDICINE        NaN        NaN   
2    M  NaN   OTHER  1994.0  INTERNAL MEDICINE        NaN        NaN   
3    M  NaN   OTHER  1994.0  INTERNAL MEDICINE        NaN        NaN   
4    M  NaN   OTHER  1994.0  INTERNAL MEDICINE        NaN        NaN   

  sec_spec_3 sec_spec_4 sec_spec_all                            org_lgl_nm  \
0        NaN        NaN          NaN   EMERGENCY MEDICINE ASSOCIATES PA PC   
1        NaN        NaN          NaN   EMERGENCY MEDICINE ASSOCIATES PA PC   
2        NaN        NaN          NaN   EMERGENCY MEDICINE ASSOCIATES PA PC   
3        NaN        NaN          NaN  SOUTHEASTERN INTENSIVIST SERVICES PC   
4        NaN        NaN          NaN  SOUTHEASTERN INTENSIVIST SERVICES PC   

     org_pac_id  num_org_mem                  adr_ln_1 adr_ln_2 ln_2_sprs  \
0  8.022915e+09        182.0        1850 TOWN CTR PKWY      NaN       NaN   
1  8.022915e+09        182.0    1701 N GEORGE MASON DR      NaN       NaN   
2  8.022915e+09        182.0  24440 STONE SPRINGS BLVD      NaN       NaN   
3  9.335152e+09        133.0   1401 JOHNSTON WILLIS DR      NaN       NaN   
4  9.335152e+09        133.0         411 W RANDOLPH RD      NaN       NaN   

                  cty  st        zip     phn_numbr hosp_afl_1  \
0              RESTON  VA  201903219  7.036899e+09     490112   
1           ARLINGTON  VA  222053610  7.035586e+09     490112   
2              DULLES  VA  201662247  5.713674e+09     490112   
3  NORTH CHESTERFIELD  VA  232354730  8.044835e+09     490112   
4            HOPEWELL  VA  238602938  8.045412e+09     490112   

       hosp_afl_lbn_1 hosp_afl_2                 hosp_afl_lbn_2 hosp_afl_3  \
0  CJW MEDICAL CENTER     210028  MEDSTAR SAINT MARY'S HOSPITAL        NaN   
1  CJW MEDICAL CENTER     210028  MEDSTAR SAINT MARY'S HOSPITAL        NaN   
2  CJW MEDICAL CENTER     210028  MEDSTAR SAINT MARY'S HOSPITAL        NaN   
3  CJW MEDICAL CENTER     210028  MEDSTAR SAINT MARY'S HOSPITAL        NaN   
4  CJW MEDICAL CENTER     210028  MEDSTAR SAINT MARY'S HOSPITAL        NaN   

  hosp_afl_lbn_3 hosp_afl_4 hosp_afl_lbn_4  hosp_afl_5 hosp_afl_lbn_5 assgn  
0            NaN        NaN            NaN         NaN            NaN     Y  
1            NaN        NaN            NaN         NaN            NaN     Y  
2            NaN        NaN            NaN         NaN            NaN     Y  
3            NaN        NaN            NaN         NaN            NaN     Y  
4            NaN        NaN            NaN         NaN            NaN     Y

Looks like we need to correct the order of the columns, like we did in the previous notebook. After we update the column order, we'll then overwrite `physician_compare_national.csv` file to reflect this change so we won't have to again going forward.

In [7]:
# create list of columns in correct order 
correct_order = ['npi', 'ind_pac_id', 'ind_enrl_id', 'lst_nm', 'frst_nm', 'mid_nm', 'suff', 
                'gndr', 'cred', 'med_sch', 'grd_yr', 'pri_spec', 'sec_spec_1', 'sec_spec_2', 
                'sec_spec_3', 'sec_spec_4', 'sec_spec_all', 'org_lgl_nm', 'org_pac_id', 
                'num_org_mem', 'adr_ln_1', 'adr_ln_2', 'ln_2_sprs', 'cty', 'st', 'zip', 
                'phn_numbr', 'hosp_afl_1', 'hosp_afl_lbn_1', 'hosp_afl_2', 'hosp_afl_lbn_2', 
                'hosp_afl_3', 'hosp_afl_lbn_3', 'hosp_afl_4', 'hosp_afl_lbn_4', 'hosp_afl_5', 
                'hosp_afl_lbn_5', 'assgn']

In [8]:
# reorder columns to original format
data = data[correct_order]

In [9]:
# overwrite original CSV file to reflect correct order (only needed to run this first time through)
# data.to_csv('physician_compare_national.csv', index=False)

# _Diving Back into the Data!_

Yesterday, we started by exploring the breakdown of physicians by state. However, I came to realize that this might be somewhat misleading as there appeared to be duplicate entries in the data (i.e. the same doctor was entered in multiple times). One of the variables though - `ind_enrl_id`, or the _Professional Enrollment ID_ - is a "unique ID for the individiaul professional enrollment that is the source for the data in the observation". This appears to be a better variable for identifying unique physicians versus the name. Let's see if this is in fact true or not with some exploration.

In [10]:
data.head()

npi  ind_pac_id      ind_enrl_id     lst_nm  frst_nm mid_nm suff  \
0  1003000126  7517003643  I20130530000085  ENKESHAFI  ARDALAN    NaN  NaN   
1  1003000126  7517003643  I20130530000085  ENKESHAFI  ARDALAN    NaN  NaN   
2  1003000126  7517003643  I20150824000105  ENKESHAFI  ARDALAN    NaN  NaN   
3  1003000126  7517003643  I20150824000105  ENKESHAFI  ARDALAN    NaN  NaN   
4  1003000126  7517003643  I20150824000105  ENKESHAFI  ARDALAN    NaN  NaN   

  gndr cred med_sch  grd_yr           pri_spec sec_spec_1 sec_spec_2  \
0    M  NaN   OTHER  1994.0  INTERNAL MEDICINE        NaN        NaN   
1    M  NaN   OTHER  1994.0  INTERNAL MEDICINE        NaN        NaN   
2    M  NaN   OTHER  1994.0  INTERNAL MEDICINE        NaN        NaN   
3    M  NaN   OTHER  1994.0  INTERNAL MEDICINE        NaN        NaN   
4    M  NaN   OTHER  1994.0  INTERNAL MEDICINE        NaN        NaN   

  sec_spec_3 sec_spec_4 sec_spec_all                            org_lgl_nm  \
0        NaN        NaN          NaN   EMERGENCY MEDICINE ASSOCIATES PA PC   
1        NaN        NaN          NaN   EMERGENCY MEDICINE ASSOCIATES PA PC   
2        NaN        NaN          NaN   EMERGENCY MEDICINE ASSOCIATES PA PC   
3        NaN        NaN          NaN  SOUTHEASTERN INTENSIVIST SERVICES PC   
4        NaN        NaN          NaN  SOUTHEASTERN INTENSIVIST SERVICES PC   

     org_pac_id  num_org_mem                  adr_ln_1 adr_ln_2 ln_2_sprs  \
0  8.022915e+09        182.0        1850 TOWN CTR PKWY      NaN       NaN   
1  8.022915e+09        182.0    1701 N GEORGE MASON DR      NaN       NaN   
2  8.022915e+09        182.0  24440 STONE SPRINGS BLVD      NaN       NaN   
3  9.335152e+09        133.0   1401 JOHNSTON WILLIS DR      NaN       NaN   
4  9.335152e+09        133.0         411 W RANDOLPH RD      NaN       NaN   

                  cty  st        zip     phn_numbr hosp_afl_1  \
0              RESTON  VA  201903219  7.036899e+09     490112   
1           ARLINGTON  VA  222053610  7.035586e+09     490112   
2              DULLES  VA  201662247  5.713674e+09     490112   
3  NORTH CHESTERFIELD  VA  232354730  8.044835e+09     490112   
4            HOPEWELL  VA  238602938  8.045412e+09     490112   

       hosp_afl_lbn_1 hosp_afl_2                 hosp_afl_lbn_2 hosp_afl_3  \
0  CJW MEDICAL CENTER     210028  MEDSTAR SAINT MARY'S HOSPITAL        NaN   
1  CJW MEDICAL CENTER     210028  MEDSTAR SAINT MARY'S HOSPITAL        NaN   
2  CJW MEDICAL CENTER     210028  MEDSTAR SAINT MARY'S HOSPITAL        NaN   
3  CJW MEDICAL CENTER     210028  MEDSTAR SAINT MARY'S HOSPITAL        NaN   
4  CJW MEDICAL CENTER     210028  MEDSTAR SAINT MARY'S HOSPITAL        NaN   

  hosp_afl_lbn_3 hosp_afl_4 hosp_afl_lbn_4  hosp_afl_5 hosp_afl_lbn_5 assgn  
0            NaN        NaN            NaN         NaN            NaN     Y  
1            NaN        NaN            NaN         NaN            NaN     Y  
2            NaN        NaN            NaN         NaN            NaN     Y  
3            NaN        NaN            NaN         NaN            NaN     Y  
4            NaN        NaN            NaN         NaN            NaN     Y

In [11]:
# get overview of the columns and dtypes within each one
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2210790 entries, 0 to 2210789
Data columns (total 38 columns):
npi               int64
ind_pac_id        int64
ind_enrl_id       object
lst_nm            object
frst_nm           object
mid_nm            object
suff              object
gndr              object
cred              object
med_sch           object
grd_yr            float64
pri_spec          object
sec_spec_1        object
sec_spec_2        object
sec_spec_3        object
sec_spec_4        object
sec_spec_all      object
org_lgl_nm        object
org_pac_id        float64
num_org_mem       float64
adr_ln_1          object
adr_ln_2          object
ln_2_sprs         object
cty               object
st                object
zip               object
phn_numbr         float64
hosp_afl_1        object
hosp_afl_lbn_1    object
hosp_afl_2        object
hosp_afl_lbn_2    object
hosp_afl_3        object
hosp_afl_lbn_3    object
hosp_afl_4        object
hosp_afl_lbn_4    object
hosp_afl_5

In [12]:
# see the percentage of observations that are NaNs in each column
round((data.isna().mean() * 100), 3)

npi                0.000
ind_pac_id         0.000
ind_enrl_id        0.000
lst_nm             0.002
frst_nm            0.001
mid_nm            24.582
suff              98.164
gndr               0.000
cred              68.976
med_sch            0.000
grd_yr             0.235
pri_spec           0.000
sec_spec_1        85.576
sec_spec_2        98.406
sec_spec_3        99.833
sec_spec_4        99.973
sec_spec_all      85.576
org_lgl_nm         8.051
org_pac_id         8.051
num_org_mem        8.051
adr_ln_1           0.000
adr_ln_2          65.159
ln_2_sprs         94.270
cty                0.000
st                 0.000
zip                0.000
phn_numbr         15.288
hosp_afl_1        26.610
hosp_afl_lbn_1    26.715
hosp_afl_2        59.499
hosp_afl_lbn_2    59.635
hosp_afl_3        76.644
hosp_afl_lbn_3    76.739
hosp_afl_4        85.746
hosp_afl_lbn_4    85.799
hosp_afl_5        90.769
hosp_afl_lbn_5    90.807
assgn              0.000
dtype: float64

Ok so we can see that there are quite a few columns that have a significant number of `NaN`s, which is something we'll have to address as we move forward. However, `ind_enrl_id` looks good to go, which is a good first step. The next thing we'll take a look out after exploring `ind_enrl_id` is how to possibly combine the first, middle and last name into one column, or if it is even necessary.

In [13]:
# see how many unique professional enrollment ID numbers there are
len(data['ind_enrl_id'].unique())

1147121

Ok so there appears to be a little over 1.1 million unique professional enrollment IDs in the data set, which is way less than the total number of observations. 

In [14]:
# combine first and last name
data['frst_nm'].str.cat(data['lst_nm'], sep=" ")

0            ARDALAN ENKESHAFI
1            ARDALAN ENKESHAFI
2            ARDALAN ENKESHAFI
3            ARDALAN ENKESHAFI
4            ARDALAN ENKESHAFI
5                THOMAS CIBULL
6                THOMAS CIBULL
7                THOMAS CIBULL
8                THOMAS CIBULL
9                THOMAS CIBULL
10               THOMAS CIBULL
11               THOMAS CIBULL
12               THOMAS CIBULL
13               THOMAS CIBULL
14               RASHID KHALIL
15               RASHID KHALIL
16               RASHID KHALIL
17               RASHID KHALIL
18               RASHID KHALIL
19               RASHID KHALIL
20            JENNIFER VELOTTA
21             KEVIN ROTHCHILD
22             KEVIN ROTHCHILD
23           FREDERICK WEIGAND
24           FREDERICK WEIGAND
25           FREDERICK WEIGAND
26             AMANDA SEMONCHE
27             AMANDA SEMONCHE
28                     DAE KIM
29            PEYMAN BENHARASH
                  ...         
2210760             LISA DRIES
2210761 

As we can see above, there are multiple cases duplicate entries are evident by combining first and last name. Now let's see how many unique names there are.

In [15]:
# combine first and last name and see how many unique names there are
len(data['frst_nm'].str.cat(data['lst_nm'], sep=" ").unique())

933045

Interesting...there are `933045` unique names but `1147121` unique professional enrollment IDs. That's a difference of over 200k+ observations! What is going on here?

In [16]:
# let's revisit how many missing values there are in the name-related columns
data[['lst_nm', 'frst_nm', 'mid_nm', 'suff']].isna().sum()

lst_nm          37
frst_nm         24
mid_nm      543463
suff       2170208
dtype: int64

In [17]:
# let's replace the NaNs with blank "" to make them a little easier to work with
data[['lst_nm', 'frst_nm']] = data[['lst_nm', 'frst_nm']].fillna(value="")

In [18]:
# let's dive into the missing values for first and last name
data[data['lst_nm'] == ""]

npi  ind_pac_id      ind_enrl_id lst_nm    frst_nm     mid_nm  \
136176   1063454130  8628071131  I20060822000324          BENJAMIN     HANBOK   
136177   1063454130  8628071131  I20060822000324          BENJAMIN     HANBOK   
177939   1083084875  4284937251  I20160125002851            TRAVIS          J   
243886   1114025665   941246326  I20050707000585             TERRY        NaN   
243887   1114025665   941246326  I20050707000585             TERRY        NaN   
298158   1134289879   941201511  I20070124000749             STEVE          C   
298159   1134289879   941201511  I20070124000749             STEVE          C   
352205   1154847614  3375802556  I20180116000499             TANYA          M   
445731   1205119435   840458907  I20120221000078         CHRISTINE          M   
445732   1205119435   840458907  I20120221000078         CHRISTINE          M   
466178   1215064852  2860663416  I20140417002113         ELIZABETH        ANN   
466179   1215064852  2860663416  I20140417002113         ELIZABETH        ANN   
466180   1215064852  2860663416  I20140417002113         ELIZABETH        ANN   
466181   1215064852  2860663416  I20140417002113         ELIZABETH        ANN   
466182   1215064852  2860663416  I20140417002113         ELIZABETH        ANN   
466183   1215064852  2860663416  I20140417002113         ELIZABETH        ANN   
506570   1225488034  2860780723  I20161007001210             SARAH          M   
555152   1255328902  4880614916  I20111207000177              SANG        NaN   
621584   1285078295  9133492952  I20170830001109              HANA        NaN   
621585   1285078295  9133492952  I20170901003155              HANA        NaN   
686317   1316017429  6103013941  I20101207000117            WALTER          B   
733084   1336129444  2365505997  I20090108000380           DANETTE        NaN   
1057235  1477816874  7618264391  I20160923000168             MANDA        NaN   
1284693  1588000723  3274830526  I20160331000662           COLLEEN          E   
1332889  1609131598   941520084  I20190422000630            KENDRA          R   
1511204  1689638744   941384515  I20080221000334            DANIEL          H   
1511205  1689638744   941384515  I20080221000334            DANIEL          H   
1515978  1689695553  4385691112  I20050405000171             SCOTT          A   
1550051  1700049269  4284872789  I20130522000411                XI        NaN   
1657935  1740768993  3173872769  I20180815000176             LOGAN       ALAN   
1714641  1770676231  1456545813  I20180129002718            ANGELA          D   
1877653  1841655669  1456655059  I20160202000675            PAMELA          A   
1889107  1851440127  8022284272  I20120112000520              SONG          H   
1895936  1851614648  5698952638  I20110531000663              GINA          D   
2076047  1932547866  4981847456  I20190531001894            ROBERT  CHRISTIAN   
2076048  1932547866  4981847456  I20190603001511            ROBERT  CHRISTIAN   
2111350  1952483950  4587826714  I20120503000468            ROBERT          N   

        suff gndr cred                                            med_sch  \
136176   NaN    M   OD                                              OTHER   
136177   NaN    M   OD                                              OTHER   
177939   NaN    M  NaN                                              OTHER   
243886   NaN    M   OD           INDIANA UNIVERSITY - SCHOOL OF OPTOMETRY   
243887   NaN    M   OD           INDIANA UNIVERSITY - SCHOOL OF OPTOMETRY   
298158   NaN    M   MD                                              OTHER   
298159   NaN    M   MD                                              OTHER   
352205   NaN    F  NaN                                              OTHER   
445731   NaN    F  NaN                                              OTHER   
445732   NaN    F  NaN                                              OTHER   
466178   NaN    F  NaN                                              OTHER   
466179   NaN 

In [19]:
data[data['frst_nm'] == ""]

npi  ind_pac_id      ind_enrl_id lst_nm frst_nm mid_nm suff  \
80900    1033381983  4183808488  I20110412000132     MA            NaN  NaN   
80901    1033381983  4183808488  I20110412000132     MA            NaN  NaN   
80902    1033381983  4183808488  I20110412000132     MA            NaN  NaN   
128852   1053726695    42532210  I20190130002955    JEN            NaN  NaN   
500235   1225268824  4385868678  I20140605002253   SHEN            NaN  NaN   
500236   1225268824  4385868678  I20140605002253   SHEN            NaN  NaN   
500237   1225268824  4385868678  I20140605002253   SHEN            NaN  NaN   
885207   1407012263  8527215920  I20120820000693    LEE              Y  NaN   
1138928  1518288307  4082929229  I20150817001415    LIU            NaN  NaN   
1141501  1518399344  6002138021  I20141202002187   VANG            NaN  NaN   
1141502  1518399344  6002138021  I20141202002187   VANG            NaN  NaN   
1304346  1588922041  7113148420  I20151019002117     LI            NaN  NaN   
1428411  1649416504  4385883222  I20130625000455     LI            NaN  NaN   
1428412  1649416504  4385883222  I20130625000455     LI            NaN  NaN   
1428413  1649416504  4385883222  I20130625000455     LI            NaN  NaN   
1428414  1649416504  4385883222  I20130625000455     LI            NaN  NaN   
1428415  1649416504  4385883222  I20130625000455     LI            NaN  NaN   
1428416  1649416504  4385883222  I20130625000455     LI            NaN  NaN   
1693799  1760644967  1951681220  I20161201001930   YANG            NaN  NaN   
1744769  1780918615  8426196981  I20091104000162     LI            NaN  NaN   
1744770  1780918615  8426196981  I20091104000162     LI            NaN  NaN   
1802680  1811350218  8325332901  I20170816002174  SMITH            NaN  NaN   
1855177  1831538214  7416241500  I20160812000332    ZHU            NaN  NaN   
2160250  1972774446  2062591282  I20080429000844  JIANG            NaN  NaN   

        gndr cred                                            med_sch  grd_yr  \
80900      F  NaN                                              OTHER  1997.0   
80901      F  NaN                                              OTHER  1997.0   
80902      F  NaN                                              OTHER  1997.0   
128852     F  NaN          LAKE ERIE COLLEGE OF OSTEOPATHIC MEDICINE  2014.0   
500235     F  NaN          UNIVERSITY OF MARYLAND SCHOOL OF MEDICINE  2009.0   
500236     F  NaN          UNIVERSITY OF MARYLAND SCHOOL OF MEDICINE  2009.0   
500237     F  NaN          UNIVERSITY OF MARYLAND SCHOOL OF MEDICINE  2009.0   
885207     F  NaN  JEFFERSON MEDICAL COLLEGE OF THOMAS JEFFERSON ...  2007.0   
1138928    F  NaN                                              OTHER  1994.0   
1141501    F  NaN                                              OTHER  2011.0   
1141502    F  NaN                                              OTHER  2011.0   
1304346    F  NaN                                              OTHER  1994.0   
1428411    F  NaN                                              OTHER  1998.0   
1428412    F  NaN                                              OTHER  1998.0   
1428413    F  NaN                                              OTHER  1998.0   
1428414    F  NaN                                              OTHER  1998.0   
1428415    F  NaN                                              OTHER  1998.0   
1428416    F  NaN                                              OTHER  1998.0   
1693799    F  NaN                                              OTHER  2005.0   
1744769    F  NaN                                              OTHER  1998.0   
1744770    F  NaN                                              OTHER  1998.0   
1802680    F  NaN                                              OTHER  2016.0   
1855177    F  NaN                                              OTHER  2005.0   
2160250    F  NaN                                              OTHER  1992.0   

                    pri_spec         se

Due to the number of observations being so low for `lst_nm` and `frst_nm` - 37 and 24, respectively - compared to the overall data set, a good tactic might be to drop them. However, when we look at the observations, they still have value as there is information regarding the school attended, year graduated, specialty, etc. so dropping them might not be the best idea. Plus none of them are missing their professional enrollment ID, so what should we do? 

Let's assign a value of `missing` as a string to any observation that does not have a first or last name. This will allow us to keep the rest of the data for the observations while also affirming that there is incomplete information regarding exactly who this physician is. 

In [20]:
# place "" with "missing" in both first and last name
data[['lst_nm', 'frst_nm']] = data[['lst_nm', 'frst_nm']].replace("", "Missing")

In [21]:
# example of missing first name and how it was replaced (look at frst_nm)
data.iloc[80900]

npi                                     1033381983
ind_pac_id                              4183808488
ind_enrl_id                        I20110412000132
lst_nm                                          MA
frst_nm                                    Missing
mid_nm                                         NaN
suff                                           NaN
gndr                                             F
cred                                           NaN
med_sch                                      OTHER
grd_yr                                        1997
pri_spec                             OPHTHALMOLOGY
sec_spec_1                                     NaN
sec_spec_2                                     NaN
sec_spec_3                                     NaN
sec_spec_4                                     NaN
sec_spec_all                                   NaN
org_lgl_nm                                     NaN
org_pac_id                                     NaN
num_org_mem                    

In [22]:
# example of missing last name and how it was replaced (look at lst_nm)
data.iloc[136176]

npi                    1063454130
ind_pac_id             8628071131
ind_enrl_id       I20060822000324
lst_nm                    Missing
frst_nm                  BENJAMIN
mid_nm                     HANBOK
suff                          NaN
gndr                            M
cred                           OD
med_sch                     OTHER
grd_yr                       2001
pri_spec                OPTOMETRY
sec_spec_1                    NaN
sec_spec_2                    NaN
sec_spec_3                    NaN
sec_spec_4                    NaN
sec_spec_all                  NaN
org_lgl_nm                    NaN
org_pac_id                    NaN
num_org_mem                   NaN
adr_ln_1           115 W 25TH AVE
adr_ln_2                      NaN
ln_2_sprs                     NaN
cty                     SAN MATEO
st                             CA
zip                     944032259
phn_numbr              6.5035e+09
hosp_afl_1                    NaN
hosp_afl_lbn_1                NaN
hosp_afl_2    

The next name related issues we have to address has to do with the middle name and suffix. 

In [23]:
data[['mid_nm', 'suff']].isna().mean()

mid_nm    0.245823
suff      0.981644
dtype: float64

Let's do what we did before and replace `NaN` in these columns with a blank `""`.

In [24]:
# fill nan with ""
data[['mid_nm', 'suff']] = data[['mid_nm', 'suff']].fillna(value="")

In [25]:
# check to see percentage of nan in middle name and suffix columns
data[['mid_nm', 'suff']].isna().mean()

mid_nm    0.0
suff      0.0
dtype: float64

In [26]:
# check percentage of nan in all name related columns
data[['lst_nm', 'frst_nm', 'mid_nm', 'suff']].isna().mean()

lst_nm     0.0
frst_nm    0.0
mid_nm     0.0
suff       0.0
dtype: float64

Ok cool now what we're going to do is combine all four of these columns into one to see if it might help us address the difference in unique IDs and names.

In [28]:
# %%timeit
# data[['frst_nm', 'mid_nm', 'lst_nm', 'suff']].apply(lambda x: ' '.join(x.astype(str)), axis=1)

In [29]:
%%timeit
data['frst_nm'].astype(str) + ' ' + data['mid_nm'].astype(str) + ' ' + data['lst_nm'].astype(str) + ' ' + data['suff'].astype(str)

1.63 s ± 308 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)


In [31]:
# create column of full names
data['full_nm'] = data['frst_nm'].astype(str) + ' ' + data['mid_nm'].astype(str) + ' ' + data['lst_nm'].astype(str) + ' ' + data['suff'].astype(str)

In [33]:
# see the first 10 full names
data['full_nm'][:10]

0    ARDALAN  ENKESHAFI 
1    ARDALAN  ENKESHAFI 
2    ARDALAN  ENKESHAFI 
3    ARDALAN  ENKESHAFI 
4    ARDALAN  ENKESHAFI 
5       THOMAS L CIBULL 
6       THOMAS L CIBULL 
7       THOMAS L CIBULL 
8       THOMAS L CIBULL 
9       THOMAS L CIBULL 
Name: full_nm, dtype: object

Ok so we've combined all the name information into one column. Let's see if it helps us at all when it comes to identifying where there is fewer names than IDs.

In [34]:
# return the total number of unique full names
len(data['full_nm'].unique())

1072060

Ok it looks like this helped increase the number of unique names! It is still less than the total number of unique IDs though as we can see below:

In [35]:
# see how many unique professional enrollment ID numbers there are
len(data['ind_enrl_id'].unique())

1147121

In [36]:
1147121 - 1072060

75061

However, the gap is not nearly as large as it was before which is a good thing. We still need to keep digging though to determine why there is this discrepency.

In [37]:
data[['full_nm', 'ind_enrl_id']].head()

full_nm      ind_enrl_id
0  ARDALAN  ENKESHAFI   I20130530000085
1  ARDALAN  ENKESHAFI   I20130530000085
2  ARDALAN  ENKESHAFI   I20150824000105
3  ARDALAN  ENKESHAFI   I20150824000105
4  ARDALAN  ENKESHAFI   I20150824000105

Luckily we don't have to look too hard as the first five rows shows us that the same physician can have multiple IDs. I hypothesize that this has to do with the primary specialty and any other secondary specialties a physician may have.

In [39]:
data.head(10)

npi  ind_pac_id      ind_enrl_id     lst_nm  frst_nm mid_nm suff  \
0  1003000126  7517003643  I20130530000085  ENKESHAFI  ARDALAN               
1  1003000126  7517003643  I20130530000085  ENKESHAFI  ARDALAN               
2  1003000126  7517003643  I20150824000105  ENKESHAFI  ARDALAN               
3  1003000126  7517003643  I20150824000105  ENKESHAFI  ARDALAN               
4  1003000126  7517003643  I20150824000105  ENKESHAFI  ARDALAN               
5  1003000134  4284706367  I20080707000385     CIBULL   THOMAS      L        
6  1003000134  4284706367  I20080707000385     CIBULL   THOMAS      L        
7  1003000134  4284706367  I20080707000385     CIBULL   THOMAS      L        
8  1003000134  4284706367  I20080707000385     CIBULL   THOMAS      L        
9  1003000134  4284706367  I20080707000385     CIBULL   THOMAS      L        

  gndr cred                                     med_sch  grd_yr  \
0    M  NaN                                       OTHER  1994.0   
1    M  NaN                                       OTHER  1994.0   
2    M  NaN                                       OTHER  1994.0   
3    M  NaN                                       OTHER  1994.0   
4    M  NaN                                       OTHER  1994.0   
5    M  NaN  UNIVERSITY OF KENTUCKY COLLEGE OF MEDICINE  2003.0   
6    M  NaN  UNIVERSITY OF KENTUCKY COLLEGE OF MEDICINE  2003.0   
7    M  NaN  UNIVERSITY OF KENTUCKY COLLEGE OF MEDICINE  2003.0   
8    M  NaN  UNIVERSITY OF KENTUCKY COLLEGE OF MEDICINE  2003.0   
9    M  NaN  UNIVERSITY OF KENTUCKY COLLEGE OF MEDICINE  2003.0   

            pri_spec sec_spec_1 sec_spec_2 sec_spec_3 sec_spec_4 sec_spec_all  \
0  INTERNAL MEDICINE        NaN        NaN        NaN        NaN          NaN   
1  INTERNAL MEDICINE        NaN        NaN        NaN        NaN          NaN   
2  INTERNAL MEDICINE        NaN        NaN        NaN        NaN          NaN   
3  INTERNAL MEDICINE        NaN        NaN        NaN        NaN          NaN   
4  INTERNAL MEDICINE        NaN        NaN        NaN        NaN          NaN   
5          PATHOLOGY        NaN        NaN        NaN        NaN          NaN   
6          PATHOLOGY        NaN        NaN        NaN        NaN          NaN   
7          PATHOLOGY        NaN        NaN        NaN        NaN          NaN   
8          PATHOLOGY        NaN        NaN        NaN        NaN          NaN   
9          PATHOLOGY        NaN        NaN        NaN        NaN          NaN   

                                          org_lgl_nm    org_pac_id  \
0                EMERGENCY MEDICINE ASSOCIATES PA PC  8.022915e+09   
1                EMERGENCY MEDICINE ASSOCIATES PA PC  8.022915e+09   
2                EMERGENCY MEDICINE ASSOCIATES PA PC  8.022915e+09   
3               SOUTHEASTERN INTENSIVIST SERVICES PC  9.335152e+09   
4               SOUTHEASTERN INTENSIVIST SERVICES PC  9.335152e+09   
5  NORTHSHORE UNIVERSITY HEALTHSYSTEM FACULTY PRA...  2.163335e+09   
6  NORTHSHORE UNIVERSITY HEALTHSYSTEM FACULTY PRA...  2.163335e+09   
7  NORTHSHORE UNIVERSITY HEALTHSYSTEM FACULTY PRA...  2.163335e+09   
8  NORTHSHORE UNIVERSITY HEALTHSYSTEM FACULTY PRA...  2.163335e+09   
9  NORTHSHORE UNIVERSITY HEALTHSYSTEM FACULTY PRA...  2.163335e+09   

   num_org_mem                     adr_ln_1 adr_ln_2 ln_2_sprs  \
0        182.0           1850 TOWN CTR PKWY      NaN       NaN   
1        182.0       1701 N GEORGE MASON DR      NaN       NaN   
2        182.0     24440 STONE SPRINGS BLVD      NaN       NaN   
3        133.0      1401 JOHNSTON WILLIS DR      NaN       NaN   
4        133.0            411 W RANDOLPH RD      NaN       NaN   
5       1286.0            2100 PFINGSTEN RD      NaN       NaN   
6       1286.0  2180 PFINGSTEN RD 1ST FLOOR      NaN         Y   
7       1286.0               777 PARK W AVE      NaN       NaN   
8       1286.0               777 PARK AVE W      NaN         Y   
9       1286.0          9600 GROSS POINT RD      NaN       NaN   

                  cty  

The first three columns - `npi`, `ind_pac_id`, and `ind_enrl_id` - provide us information regarding that particular physician's ID assigned by NPPES, PECOS, and for individual professional enrollment (which is the course for the data in that particular observation). All three are different, and we can see that for the first two physicians listed - `ENKESHAFI ARDALAN` and `CIBULL THOMAS` - that their `npi` and `ind_pac_id` are the same for all observations. However, when we look to `ENKESHAFI ARDALAN` we can see that there are two unique `ind_enrl_id`s: `I20130530000085` & `I20150824000105`.

I think I'm going to have to do a little bit more research about what exactly is going on with `ind_pac_id`. Since we haven't really done any exploration on the other two IDs, let's see if either might serve as a better unique identifier. 

In [40]:
# see how many unqiue observations are in NPI id column
len(data['npi'].unique())

1103330

In [41]:
# see how many unique observations are in PAC ID column
len(data['ind_pac_id'].unique())

1103330

As we can see above, both have the same number of unique IDs. However, there are still more unique values than there are unique names - _1,103,330_ versus _1,072,060_, respectively. This is something I'm very intrigued by but for now let's shelve it and start exploring other facets of the data. The next two columns that catch my eye are the `med_sch` and `grd_yr` columns (i.e. medical school physician attended and their graduation year). 

In [43]:
# lets see what data types we are dealing with, starting with med_sch
print('Data type for the med_sch column: {}'.format(data['med_sch'].dtype))
print('Total number of unique values in the med_sch column: {}'.format(len(data['med_sch'].unique())))

Data type for the med_sch column: object
Total number of unique values in the med_sch column: 401


So we're working with string values, and `401` unique schools. As a proxy to best address duplicate entries (as we saw, the same physican can have multiple rows thus inflating the number of instances of his/her school) we'll group by `npi` and then see the breakdown by school.

In [58]:
# ensure that when we group by NPI 
len(data.groupby('npi')['med_sch'].first())

1103330

In [60]:
# lets see what it looks like as a pandas DataFrame
pd.DataFrame(data.groupby('npi')['med_sch'].first()).head()

med_sch
npi                                                   
1003000126                                       OTHER
1003000134  UNIVERSITY OF KENTUCKY COLLEGE OF MEDICINE
1003000142                                       OTHER
1003000423                      TOLEDO MEDICAL COLLEGE
1003000480   OHIO STATE UNIVERSITY COLLEGE OF MEDICINE

I like what I'm seeing! Let's go ahead and create a `pandas` DataFrame from this so that we can analyze it further.

In [61]:
# create school count dataframe
school_count = pd.DataFrame(data.groupby('npi')['med_sch'].first()); school_count.head()

med_sch
npi                                                   
1003000126                                       OTHER
1003000134  UNIVERSITY OF KENTUCKY COLLEGE OF MEDICINE
1003000142                                       OTHER
1003000423                      TOLEDO MEDICAL COLLEGE
1003000480   OHIO STATE UNIVERSITY COLLEGE OF MEDICINE

In [63]:
# reset the index to make npi its own column
school_count = school_count.reset_index(); school_count.head()

npi                                     med_sch
0  1003000126                                       OTHER
1  1003000134  UNIVERSITY OF KENTUCKY COLLEGE OF MEDICINE
2  1003000142                                       OTHER
3  1003000423                      TOLEDO MEDICAL COLLEGE
4  1003000480   OHIO STATE UNIVERSITY COLLEGE OF MEDICINE

In [65]:
# let's get the value counts now for the schools, displaying only the top 10
school_count['med_sch'].value_counts()[:10]

OTHER                                                       589730
INDIANA UNIVERSITY SCHOOL OF MEDICINE                         7375
WAYNE STATE UNIVERSITY SCHOOL OF MEDICINE                     7297
PALMER COLLEGE CHIROPRACTIC - DAVENPORT                       6849
PHILADELPHIA COLLEGE OF OSTEOPATHIC MEDICINE                  6236
UNIVERSITY OF MINNESOTA MEDICAL SCHOOL                        6201
TEMPLE UNIVERSITY SCHOOL OF MEDICINE                          5986
OHIO STATE UNIVERSITY COLLEGE OF MEDICINE                     5986
UNIVERSITY OF ILLINOIS AT CHICAGO HEALTH SCIENCE CENTER       5665
JEFFERSON MEDICAL COLLEGE OF THOMAS JEFFERSON UNIVERSITY      5415
Name: med_sch, dtype: int64

Looks like other is the overwhelming majority of observations! To get a better idea of the distribution, let's use percentages in place of absolute counts.

In [69]:
# normalize counts so it returns a percentage
round((school_count['med_sch'].value_counts(normalize=True) * 100), 3)[:10]

OTHER                                                       53.450
INDIANA UNIVERSITY SCHOOL OF MEDICINE                        0.668
WAYNE STATE UNIVERSITY SCHOOL OF MEDICINE                    0.661
PALMER COLLEGE CHIROPRACTIC - DAVENPORT                      0.621
PHILADELPHIA COLLEGE OF OSTEOPATHIC MEDICINE                 0.565
UNIVERSITY OF MINNESOTA MEDICAL SCHOOL                       0.562
TEMPLE UNIVERSITY SCHOOL OF MEDICINE                         0.543
OHIO STATE UNIVERSITY COLLEGE OF MEDICINE                    0.543
UNIVERSITY OF ILLINOIS AT CHICAGO HEALTH SCIENCE CENTER      0.513
JEFFERSON MEDICAL COLLEGE OF THOMAS JEFFERSON UNIVERSITY     0.491
Name: med_sch, dtype: float64

In [72]:
# let's look a little bit further down the list (and return to counts)
school_count['med_sch'].value_counts(normalize=False)[10:20]

NEW YORK MEDICAL COLLEGE                                     5390
UNIVERSITY OF MICHIGAN MEDICAL SCHOOL                        5338
GEORGETOWN UNIVERSITY OF MEDICINE                            5111
UNIVERSITY OF ALABAMA SCHOOL OF MEDICINE                     4976
UNIVERSITY OF KANSAS SCHOOL OF MEDICINE                      4896
MEDICAL COLLEGE OF GEORGIA                                   4882
UNIVERSITY OF TENNESSEE COLLEGE OF MEDICINE                  4823
UNIVERSITY OF CINCINNATI COLLEGE OF MEDICINE                 4771
UNIVERSITY OF TEXAS SOUTHWESTERN MEDICAL SCHOOL AT DALLAS    4708
UNIVERSITY OF TEXAS MEDICAL BRANCH AT GALVESTON              4690
Name: med_sch, dtype: int64

In [73]:
# a little bit further...
school_count['med_sch'].value_counts(normalize=False)[20:30]

NEW YORK UNIVERSITY SCHOOL OF MEDICINE                          4630
UNIVERSITY OF WASHINGTON SCHOOL OF MEDICINE                     4524
LOUISIANA STATE UNIVERSITY SCHOOL OF MEDICINE IN NEW ORLEANS    4483
MEDICAL COLLEGE OF WISCONSIN                                    4471
UNIVERSITY OF IOWA COLLEGE OF MEDICINE                          4410
BAYLOR COLLEGE OF MEDICINE                                      4344
UNIVERSITY OF WISCONSIN MEDICAL SCHOOL                          4262
BOSTON UNIVERSITY SCHOOL OF MEDICINE                            4203
NORTHWESTERN UNIVERSITY MEDICAL SCHOOL                          4168
TUFTS UNIVERSITY SCHOOL OF MEDICINE                             4152
Name: med_sch, dtype: int64

In [75]:
data.tail()

npi  ind_pac_id      ind_enrl_id     lst_nm   frst_nm mid_nm  \
2210785  1992999825   143414284  I20101102000933  DESCHENES  GEOFFREY      R   
2210786  1992999825   143414284  I20101102000933  DESCHENES  GEOFFREY      R   
2210787  1992999825   143414284  I20101102000933  DESCHENES  GEOFFREY      R   
2210788  1992999825   143414284  I20101102000933  DESCHENES  GEOFFREY      R   
2210789  1992999825   143414284  I20101102000933  DESCHENES  GEOFFREY      R   

        suff gndr cred                                            med_sch  \
2210785         M  NaN  JEFFERSON MEDICAL COLLEGE OF THOMAS JEFFERSON ...   
2210786         M  NaN  JEFFERSON MEDICAL COLLEGE OF THOMAS JEFFERSON ...   
2210787         M  NaN  JEFFERSON MEDICAL COLLEGE OF THOMAS JEFFERSON ...   
2210788         M  NaN  JEFFERSON MEDICAL COLLEGE OF THOMAS JEFFERSON ...   
2210789         M  NaN  JEFFERSON MEDICAL COLLEGE OF THOMAS JEFFERSON ...   

         grd_yr        pri_spec sec_spec_1 sec_spec_2 sec_spec_3 sec_spec_4  \
2210785  2005.0  OTOLARYNGOLOGY        NaN        NaN        NaN        NaN   
2210786  2005.0  OTOLARYNGOLOGY        NaN        NaN        NaN        NaN   
2210787  2005.0  OTOLARYNGOLOGY        NaN        NaN        NaN        NaN   
2210788  2005.0  OTOLARYNGOLOGY        NaN        NaN        NaN        NaN   
2210789  2005.0  OTOLARYNGOLOGY        NaN        NaN        NaN        NaN   

        sec_spec_all                     org_lgl_nm    org_pac_id  \
2210785          NaN  VIRGINIA MASON MEDICAL CENTER  9.830003e+09   
2210786          NaN  VIRGINIA MASON MEDICAL CENTER  9.830003e+09   
2210787          NaN  VIRGINIA MASON MEDICAL CENTER  9.830003e+09   
2210788          NaN  VIRGINIA MASON MEDICAL CENTER  9.830003e+09   
2210789          NaN  VIRGINIA MASON MEDICAL CENTER  9.830003e+09   

         num_org_mem                adr_ln_1                       adr_ln_2  \
2210785        776.0        19116 33RD AVE W        VIRGINIA MASON LYNNWOOD   
2210786        776.0           925 SENECA ST        VIRGINIA MASON HOSPITAL   
2210787        776.0            1100 9TH AVE                            NaN   
2210788        776.0   4575 SANDPOINT WAY NE                            NaN   
2210789        776.0  4575 SAND POINT WAY NE  108 VIRGINIA MASON SAND POINT   

        ln_2_sprs       cty  st        zip     phn_numbr hosp_afl_1  \
2210785       NaN  LYNNWOOD  WA  980364706  4.257128e+09     500005   
2210786       NaN   SEATTLE  WA  981012742  2.062237e+09     500005   
2210787       NaN   SEATTLE  WA  981012756           NaN     500005   
2210788         Y   SEATTLE  WA  981053999  2.065258e+09     500005   
2210789       NaN   SEATTLE  WA  981053999  2.065258e+09     500005   

                        hosp_afl_lbn_1 hosp_afl_2 hosp_afl_lbn_2 hosp_afl_3  \
2210785  VIRGINIA MASON MEDICAL CENTER        NaN            NaN        NaN   
2210786  VIRGINIA MASON MEDICAL CENTER        NaN            NaN        NaN   
2210787  VIRGINIA MASON MEDICAL CENTER        NaN            NaN        NaN   
2210788  VIRGINIA MASON MEDICAL CENTER        NaN            NaN        NaN   
2210789  VIRGINIA MASON MEDICAL CENTER        NaN            NaN        NaN   

        hosp_afl_lbn_3 hosp_afl_4 hosp_afl_lbn_4  hosp_afl_5 hosp_afl_lbn_5  \
2210785            NaN        NaN            NaN         NaN            NaN   
2210786            NaN        NaN            NaN         NaN            NaN   
2210787            NaN        NaN            NaN         NaN            NaN   
2210788            NaN        NaN            NaN         NaN            NaN   
2210789            NaN        NaN            NaN         NaN            NaN   

        assgn                full_nm  
2210785     Y  GEOFFREY R DESCHENES   
2210786     Y  GEOFFREY R DESCHENES   
2210787     Y  GEOFFREY R DESCHENES   
2210788     Y  GEOFFREY R DESCHENES   
2210789     Y  GEOFFREY R DESCHENES

In [77]:
# slight modification to column order, adding full name after other name columns
new_order = ['npi', 'ind_pac_id', 'ind_enrl_id', 'lst_nm', 'frst_nm', 'mid_nm', 'suff', 'full_nm', 
             'gndr', 'cred', 'med_sch', 'grd_yr', 'pri_spec', 'sec_spec_1', 'sec_spec_2', 
             'sec_spec_3', 'sec_spec_4', 'sec_spec_all', 'org_lgl_nm', 'org_pac_id', 
             'num_org_mem', 'adr_ln_1', 'adr_ln_2', 'ln_2_sprs', 'cty', 'st', 'zip', 
             'phn_numbr', 'hosp_afl_1', 'hosp_afl_lbn_1', 'hosp_afl_2', 'hosp_afl_lbn_2', 
             'hosp_afl_3', 'hosp_afl_lbn_3', 'hosp_afl_4', 'hosp_afl_lbn_4', 'hosp_afl_5', 
             'hosp_afl_lbn_5', 'assgn']

# assign new column order to data
data = data[new_order]

In [78]:
# check it to make sure it worked
data.head()

npi  ind_pac_id      ind_enrl_id     lst_nm  frst_nm mid_nm suff  \
0  1003000126  7517003643  I20130530000085  ENKESHAFI  ARDALAN               
1  1003000126  7517003643  I20130530000085  ENKESHAFI  ARDALAN               
2  1003000126  7517003643  I20150824000105  ENKESHAFI  ARDALAN               
3  1003000126  7517003643  I20150824000105  ENKESHAFI  ARDALAN               
4  1003000126  7517003643  I20150824000105  ENKESHAFI  ARDALAN               

               full_nm gndr cred med_sch  grd_yr           pri_spec  \
0  ARDALAN  ENKESHAFI     M  NaN   OTHER  1994.0  INTERNAL MEDICINE   
1  ARDALAN  ENKESHAFI     M  NaN   OTHER  1994.0  INTERNAL MEDICINE   
2  ARDALAN  ENKESHAFI     M  NaN   OTHER  1994.0  INTERNAL MEDICINE   
3  ARDALAN  ENKESHAFI     M  NaN   OTHER  1994.0  INTERNAL MEDICINE   
4  ARDALAN  ENKESHAFI     M  NaN   OTHER  1994.0  INTERNAL MEDICINE   

  sec_spec_1 sec_spec_2 sec_spec_3 sec_spec_4 sec_spec_all  \
0        NaN        NaN        NaN        NaN          NaN   
1        NaN        NaN        NaN        NaN          NaN   
2        NaN        NaN        NaN        NaN          NaN   
3        NaN        NaN        NaN        NaN          NaN   
4        NaN        NaN        NaN        NaN          NaN   

                             org_lgl_nm    org_pac_id  num_org_mem  \
0   EMERGENCY MEDICINE ASSOCIATES PA PC  8.022915e+09        182.0   
1   EMERGENCY MEDICINE ASSOCIATES PA PC  8.022915e+09        182.0   
2   EMERGENCY MEDICINE ASSOCIATES PA PC  8.022915e+09        182.0   
3  SOUTHEASTERN INTENSIVIST SERVICES PC  9.335152e+09        133.0   
4  SOUTHEASTERN INTENSIVIST SERVICES PC  9.335152e+09        133.0   

                   adr_ln_1 adr_ln_2 ln_2_sprs                 cty  st  \
0        1850 TOWN CTR PKWY      NaN       NaN              RESTON  VA   
1    1701 N GEORGE MASON DR      NaN       NaN           ARLINGTON  VA   
2  24440 STONE SPRINGS BLVD      NaN       NaN              DULLES  VA   
3   1401 JOHNSTON WILLIS DR      NaN       NaN  NORTH CHESTERFIELD  VA   
4         411 W RANDOLPH RD      NaN       NaN            HOPEWELL  VA   

         zip     phn_numbr hosp_afl_1      hosp_afl_lbn_1 hosp_afl_2  \
0  201903219  7.036899e+09     490112  CJW MEDICAL CENTER     210028   
1  222053610  7.035586e+09     490112  CJW MEDICAL CENTER     210028   
2  201662247  5.713674e+09     490112  CJW MEDICAL CENTER     210028   
3  232354730  8.044835e+09     490112  CJW MEDICAL CENTER     210028   
4  238602938  8.045412e+09     490112  CJW MEDICAL CENTER     210028   

                  hosp_afl_lbn_2 hosp_afl_3 hosp_afl_lbn_3 hosp_afl_4  \
0  MEDSTAR SAINT MARY'S HOSPITAL        NaN            NaN        NaN   
1  MEDSTAR SAINT MARY'S HOSPITAL        NaN            NaN        NaN   
2  MEDSTAR SAINT MARY'S HOSPITAL        NaN            NaN        NaN   
3  MEDSTAR SAINT MARY'S HOSPITAL        NaN            NaN        NaN   
4  MEDSTAR SAINT MARY'S HOSPITAL        NaN            NaN        NaN   

  hosp_afl_lbn_4  hosp_afl_5 hosp_afl_lbn_5 assgn  
0            NaN         NaN            NaN     Y  
1            NaN         NaN            NaN     Y  
2            NaN         NaN            NaN     Y  
3            NaN         NaN            NaN     Y  
4            NaN         NaN            NaN     Y

In [79]:
# export current data to CSV for future analysis
data.to_csv('physician_compare_national-updates-1.csv', index=False)